### ISSUE TITLE
How to get the logs of my provider controller

### ISSUE THREAD

**Issue by huchen2021** [kubernetes-sigs/cluster-api#5058]
**What steps did you take and what happened:**
[A clear and concise description on how to REPRODUCE the bug.]

**What did you expect to happen:**

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

**Environment:**

- Cluster-api version:
- Minikube/KIND version:
- Kubernetes version: (use `kubectl version`):
- OS (e.g. from `/etc/os-release`):

/kind bug
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]
 ```
I am developing a provider, let's say the name is "cluster-api-provider-my".
Then I write mymachine_controller.go, Here is my Reconcile function 
func (r *MyMachineReconciler) Reconcile(ctx context.Context, req ctrl.Request) (_ ctrl.Result, reterr error) {
	logger := log.FromContext(ctx).WithValues("namespace", req.Namespace, "MyMachine", req.Name)
        ........
        logger.Info("start to work")           ----  Line 1
        logger.Error(errors.New("just to see if it can print out"), "start to work")     -- Line 2
       ........
       if err != nil {
		logger.Error(err, "Fetch machine failed")                ---  Line 3
		return ctrl.Result{}, err
	}
        ........
}

When I debug this provider, I get the log of pod my-controller-manager-5cff9bff45-2k5w2
kubectl logs -n my-system my-controller-manager-5cff9bff45-2k5w2 --kubeconfig /tmp/mgmt.conf -c manager

I found only the output of line3 is print, line 1 and line2 are not print.
Anyone knows how to get the whole log of Reconcile function  in  mymachine_controller.go? 


```

**Comment by fabriziopandini** [comment]
I'm not sure theCAPI team can help here.
My first take on this issue is that this isn't a CAPI problem, but a problem in your code (code flow is hard to follow in the example due to a messy indentation), or in the logger implementation your are using, or eventually a controller runtime problem even if I tend to exclude this last option. 
FYI in CAPI we are using klog, and I never observer a similar problem.

**Comment by huchen2021** [comment]
> I'm not sure theCAPI team can help here.
> My first take on this issue is that this isn't a CAPI problem, but a problem in your code (code flow is hard to follow in the example due to a messy indentation), or in the logger implementation your are using, or eventually a controller runtime problem even if I tend to exclude this last option.
> FYI in CAPI we are using klog, and I never observed a similar problem in my tests.

I also klog, same issue.

**Comment by vincepri** [comment]
Have you setup the logger in the main.go?

**Comment by vincepri** [comment]
I'll echo what @fabriziopandini said above though, we can try to help as much as possible, although we usually can't support code outside of this repository.
